File: jupyter2/ps/function/in_circle/go/in_circle.go
package main
import "fmt"

/** begin my answer */

/** begin hidden */
func inCircle(x,y,r float64) bool {
    return x*x + y*y <= r*r
}
/** end hidden */
/** end my answer */
func main() {
	if !inCircle(1, 1, 2) { panic("wrong") }
	if !inCircle(3, 4, 6) { panic("wrong") }
	if  inCircle(3, 0, 2) { panic("wrong") }
	fmt.Println("OK")
}

File: jupyter2/ps/function/in_circle/ml/in_circle.ml
(** begin my answer *)

(** begin hidden *)
let in_circle x y r = x *. x +. y *. y <= r *. r;;
(** end hidden *)
(** end my answer *)

let main () =
  assert (in_circle 1.0 1.0 2.0);
  assert (in_circle 3.0 4.0 6.0);
  assert (not (in_circle 3.0 0.0 2.0));
  Printf.printf "OK\n"
;;

main()
